trigger/wsclient/trigger.go

```diff
@@ -38,32 +46,111 @@ type Trigger struct {
 // New implements trigger.Factory.New
 func (*Factory) New(config *trigger.Config) (trigger.Trigger, error) {
 	s := &Settings{}
-	err := metadata.MapToStruct(config.Settings, s, true)
+	/*err := metadata.MapToStruct(config.Settings, s, true)
 	if err != nil {
 		return nil, err
-	}
+	}*/
 
 	return &Trigger{settings: s, config: config}, nil
 }
 
 // Initialize initializes the trigger
 func (t *Trigger) Initialize(ctx trigger.InitContext) error {
 	t.logger = ctx.Logger()
-	urlSetting := t.config.Settings["url"]
-	if urlSetting == nil || urlSetting.(string) == "" {
-		return fmt.Errorf("server url not provided")
+	urlstring, err := coerce.ToString(t.config.Settings["url"])
+	// populate headers
+	header := make(http.Header)
+	headerObject, err := coerce.ToObject(t.config.Settings["headers"])
+	if err != nil {
+		// queryparam is array as per FE
+		headerArray, err := coerce.ToArray(t.config.Settings["headers"])
+		if err != nil {
+			return err
+		}
+		for _, val := range headerArray {
+			qparamMap := val.(map[string]interface{})
+			header[qparamMap["parameterName"].(string)] = []string{qparamMap["value"].(string)}
```

Comment:
I don't think we need this logic anymore base on our discussion, Let's cleanup

Reply:
done